### PR TITLE
Update to imgui v1.52

### PR DESCRIPTION
**Pull request by ThisDrunkDane**
I've updated to v1.52 of dear imgui
Also moved some stuff around to match the original imgui.h

### CODE REVIEW

**Comment by Extrawurst** [comment]
Hi @ThisDrunkDane thanks for the work. Can you please make sure the Travis-CI build is green otherwise I cannot merge this.

File: cimgui/cimgui.h [hunk 1]
```diff
@@ -457,10 +472,12 @@ CIMGUI_API void             igGetWindowSize(struct ImVec2* out);
...
 CIMGUI_API float            igGetWindowHeight();
 CIMGUI_API bool             igIsWindowCollapsed();
+CIMGUI_API bool             igIsWindowAppearing();
 CIMGUI_API void             igSetWindowFontScale(float scale);
 
-CIMGUI_API void             igSetNextWindowPos(CONST struct ImVec2 pos, ImGuiCond cond);
-CIMGUI_API void             igSetNextWindowPosCenter(ImGuiCond cond);
+CIMGUI_API void             igSetNextWindowPos(CONST struct ImVec2 pos, ImGuiCond cond, CONST struct ImVec2 pivot);
+//Is going to be obsolete
+//CIMGUI_API void             igSetNextWindowPosCenter(ImGuiCond cond);
```
Comment (Extrawurst) [inline cimgui/cimgui.h]: Let us not use comments for obsoletion. Please remove the code instead of commenting it

File: cimgui/cimgui.h [hunk 2]
```diff
@@ -536,12 +553,15 @@ CIMGUI_API void             igSetCursorPosY(float y);
 CIMGUI_API void             igGetCursorStartPos(struct ImVec2* pOut);
 CIMGUI_API void             igGetCursorScreenPos(struct ImVec2* pOut);
 CIMGUI_API void             igSetCursorScreenPos(CONST struct ImVec2 pos);
-CIMGUI_API void             igAlignFirstTextHeightToWidgets();
+//Is going to be obsolete
+//CIMGUI_API void             igAlignFirstTextHeightToWidgets();
```
Comment (Extrawurst) [inline cimgui/cimgui.h]: same here.

File: cimgui/cimgui.h [hunk 3]
```diff
@@ -536,12 +553,15 @@ CIMGUI_API void             igSetCursorPosY(float y);
...
+CIMGUI_API void            igAlignTextToFramePadding();
 CIMGUI_API float            igGetTextLineHeight();
 CIMGUI_API float            igGetTextLineHeightWithSpacing();
 CIMGUI_API float            igGetItemsLineHeightWithSpacing();
 
 //Columns
+// You can also use SameLine(pos_x) for simplified columns. The columns API is still work-in-progress and rather lacking.
```
Comment (Extrawurst) [inline cimgui/cimgui.h]: same here..

File: cimgui/cimgui.h [hunk 4]
```diff
@@ -718,9 +740,13 @@ CIMGUI_API void             igLogText(CONST char* fmt, ...);
...
 
+// Styles
+CIMGUI_API void              igStyleColorsClassic(struct ImGuiStyle* dst);
+
 // Utilities
-CIMGUI_API bool             igIsItemHovered();
-CIMGUI_API bool             igIsItemRectHovered();
+CIMGUI_API bool             igIsItemHovered(ImGuiHoveredFlags flags);
+//Is going to be obsolete
+//CIMGUI_API bool             igIsItemRectHovered();
```
Comment (Extrawurst) [inline cimgui/cimgui.h]: dito

File: cimgui/cimgui.h [hunk 5]
```diff
@@ -731,11 +757,12 @@ CIMGUI_API void             igGetItemRectMax(struct ImVec2* pOut);
 CIMGUI_API void             igGetItemRectSize(struct ImVec2* pOut);
 CIMGUI_API void             igSetItemAllowOverlap();
 CIMGUI_API bool             igIsWindowFocused();
-CIMGUI_API bool             igIsWindowHovered();
-CIMGUI_API bool             igIsWindowRectHovered();
+CIMGUI_API bool             igIsWindowHovered(ImGuiHoveredFlags falgs);
+//Is going to be obsolete
+//CIMGUI_API bool             igIsWindowRectHovered();
```
Comment (Extrawurst) [inline cimgui/cimgui.h]: dito

**Comment by ThisDrunkDane** [comment]
There we go @Extrawurst

**Comment by Extrawurst** [comment]
Thank you so much!

**Pull request merged**